provisioner/inspec/provisioner.go

```diff
@@ -0,0 +1,556 @@
+package inspec
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"os/user"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"unicode"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/config"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/template/interpolate"
+)
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+	ctx                 interpolate.Context
+
+	// The command to run inspec
+	Command    string
+	SubCommand string
+
+	// Extra options to pass to the inspec command
+	ExtraArguments []string `mapstructure:"extra_arguments"`
+	InspecEnvVars  []string `mapstructure:"inspec_env_vars"`
+
+	// The profile to execute.
+	Profile              string   `mapstructure:"profile"`
+	AttributesDirectory  string   `mapstructure:"attributes_directory"`
+	AttributesFiles      []string `mapstructure:"attributes"`
+	Backend              string   `mapstructure:"backend"`
+	User                 string   `mapstructure:"user"`
+	Host                 string   `mapstructure:"host"`
+	LocalPort            string   `mapstructure:"local_port"`
```

Comment:
uint?

Reply:
for the port ?

Reply:
yes

Reply:
What is the exact reason behind this suggestion. Just that I can understand your thought process better, since the ansible provisioner does the same thing.